Add gdi09

fruestueck opened the pull request:
Add mission 9 "[Secure the Danube](https://cnc.fandom.com/wiki/Secure_the_Danube)" for the Tiberian Dawn GDI campaign (#4988).

The Lua script includes parts from [scg07ea](https://github.com/OpenRA/OpenRA/pull/11090) and [scb09ea](https://github.com/OpenRA/OpenRA/pull/11306).

Have fun testing! : )

~Original behavior:
```
+ Send one Gunboat (three in total) for each...
 * GDI Units near (CellTriggers) North NodBunker
 * First NodBunker is destroyed
 * Second NodBunker is destroyed

+ If a gunboat reaches its destination (West side of Map)
 * it will continue its patrol horizontally
 * despite the MissionGoal, the game is NOT over.

+ gunboats don't attack and get destroyed by one obelisk attack

+ A chinook is provided as soon as the south NodBunkes (gun3, gun4) get destroyed
 
+ Player WINs only when all Nod structures and units are destroyed.

+ Kill all four sam sites to gain airstrike support power

+ church does not spawn a crate (should we?)
```

Changes:
- Prerequesits for e4, ltnk & arty reduced to ~~techlevel.low~~ ~afld and barracks respectivly
- A `PatrolProductionQueue` and `ReplenishPatrolUnit()` function added for HandofNod.
OnDeath grd1 and grd2 Units will be rebuild and sent back to patrol
- For the Nod1 to 5 and Auto1 to 8 AttackWaves the lua-script (from mission scb09ea) is just grabbing available (`IsIdle`) Untis. As the Airfield and HandofNod produces these units by random, some attack waves may be incomplete at different playthrough.
This could be changed to spawn the required units at GameStart. Alternatively the required Units could be built with a queue similar to `ReplenishPatrolUnit()`. This would take much more effort though and would be a possible feature PR.
- Idling units in the woods are patrolling their starting location. This prevents them from being picked for an `AttackWave`
- A gunboat is spawned, once for each GunGroup (North & Sout of the river) beeing cleared.
In the original, a gunboat was spawned on entering TriggerCells next to the North Guns, and two times for killing the first two guns. I want to wait for comments on this before changing it. Also I didn't know how to translate eg. cell 2213 from the ini into a CPos.


Import log:

```
-Details----------------------------------------------------
Name   : GDI Mission 9, Hungary
Mission: scg09ea.ini
Source : CNC Gold
Tool   : OpenRA.Utility.exe, bleed 19.07.2019
Reference: https://cnc.fandom.com/wiki/Secure_the_Danube
-Result-----------------------------------------------------
Ignoring unknown MaxBuilding=150 for player BadGuy
Ignoring unknown MaxUnit=150 for player BadGuy
Ignoring unknown Credits=30 for player BadGuy
Ignoring unknown Edge=North for player BadGuy
Ignoring unknown MaxBuilding=150 for player Neutral
Ignoring unknown MaxUnit=150 for player Neutral
Ignoring unknown Edge=North for player Neutral
Ignoring unknown Credits=0 for player Neutral
Ignoring unknown MaxBuilding=150 for player GoodGuy
Ignoring unknown MaxUnit=341 for player GoodGuy
Ignoring unknown Credits=40 for player GoodGuy
Ignoring unknown Edge=East for player GoodGuy
scg09ea.oramap saved.
```

Possible TiberiumTreeCreation adjustment after 16827 is merged.

---
Updates:
The following corrections were made:

- Updated map.png (320 -> 360px width, visually not changed ingame)
- Bridge made undestroyable
- Airstrike SquadSize reduced to 2
- SamSite spawn e1 on death
- Gunboat health reduction for Obelisk to one-hit (vanilla)

RoosterDragon committed:
Use SegmentStream.CreateWithoutOwningStream to avoid reading data into memory.

To avoid creating copied data in memory (e.g. via MemoryStream), this method can be used to reference offsets on files on disk, reducing memory requirements.

pchote committed:
Add support for custom SupportPowerInstances.

pchote committed:
Rename *Time to *Ticks and increase internal resolution.

pchote committed:
Allow support powers to override the icon overlay/tooltip labels.

tovl committed:
Allow units to give way when path is blocked by oncoming unit.

abcdefg30 committed:
Use a global script in Tiberian Dawn

abcdefg30 committed:
Fix broken reinforcements on gdi04 and limit them to 3 waves

tovl committed:
Add option for airlift to arrive from player baseline.

tovl committed:
Fix airlift direction in isometric mods.

tovl committed:
Make airlift landing direction configurable.

dragunoff committed:
Add restart button to main menu for single player (missions, skirmish, replays)

dragunoff committed:
Always prompt the user to confirm when leaving a game

RoosterDragon committed:
Add UnionRectangles extension method.

dragunoff committed:
Add selection tiers as inheritable templates

IceReaper committed:
Fixed incorrect sequence frame amount checks using Stride.

teinarss committed:
Cleanup in MapEditorTabsLogic

teinarss committed:
Added Undo Redo to editor

abcdefg30 committed:
Remove 'AbilityInsufficientPower' from cnc's notifications

abcdefg30 committed:
Add 'AirstrikeReady' to 'DisablePrefixes'

abcdefg30 committed:
Remove the bogus 'GameSaved' speech notification definition

Commit:
Fix destroyed truck escaping ra mission sarin-gas-1

reaperrr committed:
Cache IPathFinder in Mobile at creation

Avoiding look-ups on every move order,
as well as reducing line lengths.

tovl committed:
Allow additional actor IDs to be send with orders.

pchote committed:
Add EditorActorCheckbox plumbing.

pchote committed:
Add "Spawn Child Actor" checkbox in the map editor actor properties.

pchote committed:
Add "Deployed" checkbox in the map editor actor properties.

SoScared committed:
Redo buildpaletteorder for RA vehicles

Restructure RA's vehicle build tab. More streamlined and better looking.

SoScared committed:
Redo buildpaletteorder for D2K buildings

SoScared committed:
Add map Climax to RA map pool

pchote committed:
Add GrantPrerequisiteChargeDrainPower and DrainPrerequisitePowerOnDamage.

These traits implement the Firestorm defense charge/drain logic.

pchote committed:
Fix DamageModifier crashes when an actor is demolished.

Demolish calls GetDamageModifier with a null Damage.

tovl committed:
Prevent bogus attackmove on take-off.

tovl committed:
Refactor unreserve actions.

tovl committed:
Add missing target line to aircraft taking off from resupplying.

tovl committed:
Let CreateGroup use ExtraActors field.

tovl committed:
Clean up usage of CancelActivity.

tovl committed:
Allow queued structure rallypoints.

pchote committed:
Change IPlaceBuildingDecorationInfo to use annotations.

pchote committed:
Add annotation support to IOrderGenerator.

tovl committed:
Increase IMove.MoveTo call flexibility.

abcdefg30 committed:
Add an XAxisTicksPerLabel property to LineGraphWidget

abcdefg30 committed:
Replace "$/min" by "Income" and increase graph update rates

pchote committed:
Drop invalid power references when loading save games.

pchote committed:
Expire invalid instances from the SupportPowerBotModule cache.

abcdefg30 committed:
Fix potentially bogus usages of OnAllRemovedFromWorld

pchote committed:
Fix player viewport saving for non-spectators.

PunkPun committed:
RA balance changes for September 2019

abcdefg30 committed:
Remove selling from Infilitration

It is weird, unsatisfying for the player and inconsistent with the rest of our missions

obrakmann committed:
Fix idling aircraft on Intervention

dragunoff committed:
Add duplicate hotkey tracking with a boolean property on the definition

teinarss committed:
Add a trait PlayerRadarTerrain to track explored terrain

fruestueck commented:
The following corrections were made:

- Updated map.png (320 -> 360px width, visually not changed ingame)
- Bridge made undestroyable
- Airstrike `SquadSize` reduced to `2`
- SamSite spawn `e1` on death

There is an issue that may come up:
If the squad on the south island dies before killing both gunturrets, they can only be killed with airstrikes later on. As this goal (killing north and south gun groups) is optional, it should not count against the win conditions.
I did not test it in the original and guess that it requires all units to be destroyed. The issue is a minor one but could be thought about.

pchote committed:
Convert TerrainGeometryOverlay to IRenderAboveShroud.

abcdefg30 committed:
Fix reinforcements on the ts shellmap not attack moving

abcdefg30 committed:
Convert all spaces to tabs in the ts shellmap script

RoosterDragon committed:
Implement IEquatable on structs.

Any struct which overrides object.Equals(object obj) should implement IEquatable<T> to gain a more efficient Equals(T other) overload. This overload will be used by hashing collections like Dictionary which enables them to check equality without boxing the struct.

pchote committed:
Migrate traits to IRenderAnnotations.

pchote committed:
Add IRenderAnnotations(WhenSelected) interfaces.

vesuvian committed:
feat: Using a glDebugMessageCallback instead of glGetError on devices that support it

pchote committed:
Replace scaleSizeWithZoom with SpriteAnnotation.

pchote committed:
Draw annotations using the UI renderers.

pchote committed:
Rename TextRenderable to TextAnnotationRenderable.

pchote committed:
Rename DetectionCircleRenderable to DetectionCircleAnnotationRenderable.

pchote committed:
Rename RangeCircleRenderable to RangeCircleAnnotationRenderable.

pchote committed:
Rename SelectionBarsRenderable to SelectionBarsAnnotationRenderable.

pchote committed:
Rename SelectionBoxRenderable to SelectionBoxAnnotationRenderable.

tovl committed:
Skip rally point if order is queued after resupply.

pchote committed:
Add IEffectAnnotation interface.

PunkPun committed:
Make laser fence dissable on emp

PunkPun committed:
Make EMP dissable detection of mobile sensor array

PunkPun committed:
Make EMP dissable subterranean units while they're underground

PunkPun committed:
Fix MRV not auto targeting vehicles

tovl committed:
Make locomotor cache and nudging logic aware of mobile trait status.

tovl committed:
Fix deployed units being nudgeable.

teinarss committed:
Production should set SubCellInit

abcdefg30 committed:
Fix the expansion mcv in soviet05 being transported off the map

abcdefg30 committed:
Remove the now unnecessary Helper refinery

abcdefg30 committed:
Increase the SearchFromProcRadius radius in soviet05

tovl committed:
Clean up nudging code.

abcdefg30 committed:
Fix passengers idling on the ts shellmap

abcdefg30 committed:
Fix unarmed units idling on the ts shellmap

abcdefg30 committed:
Revert the search radius decrease in TS

abcdefg30 committed:
Fix setting the position of the wrong actor

abcdefg30 committed:
Add an ICreationActivity interface

abcdefg30 committed:
Revert the search radius decrease in D2k

teinarss committed:
Add check to see if transport is dead to UnreserveSpace

PunkPun committed:
Make mine targetable on attack everything stance for AutoTargetGround

PunkPun committed:
Fix ctank and stank building faster then they should

abcdefg30 committed:
Fix actors returning fire at invisible attackers

abcdefg30 committed:
Don't allow movement for actors without IMove

However, this does not check if any existing IMove traits are enabled.

teinarss committed:
Rename duplicated actors

abcdefg30 committed:
Fix/Update the patrols on nod07c

abcdefg30 committed:
Make orcas buildable on nod07c

teinarss committed:
Mark cells that have changed MovementType as dirty

teinarss committed:
Remove CanEnterCell from OccupiedCells

dragunoff committed:
Set duplicates flag for hotkeys in HotkeyManager

abcdefg30 committed:
Throw a lua exception when setting an owner to null/nil

tovl committed:
Add separate condition to mobile for disabling nudging only.

cjshmyr committed:
Added TooltipName as an actor Lua property

abcdefg30 committed:
Prevent chronoshifting an empty selection

pchote committed:
Add Engine.LaunchWrapper launch argument for mod switching.

pchote committed:
Reset environment variables before switching mods.

pchote committed:
Remove double-negative from appimage wrapper.

abcdefg30 committed:
Fix double clicking a save in the save game dialogue loading it

abcdefg30 committed:
Add SearchFromProcRadius to TD missions that need it

abcdefg30 committed:
Remove unnecessary SearchFromHarvesterRadius overwrites from TD missions

pchote committed:
Update packaged mono to 6.4.0.

pchote committed:
Compile using Mono 6.4.0.

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09.lua:
```diff
@@ -0,0 +1,216 @@
+	
+	Trigger.AfterDelay(DateTime.Minutes(1), function() StartWaves() end) -- DateTime.Minutes(1)
+	Trigger.AfterDelay(DateTime.Minutes(3), function() ProduceInfantry(handofnod) end) -- DateTime.Minutes(3)
+	Trigger.AfterDelay(DateTime.Minutes(3), function() ProduceVehicle(nodairfield) end) -- DateTime.Minutes(3)
```
The comments here don't look necessary.

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09.lua:
```diff
@@ -0,0 +1,216 @@
+	Trigger.AfterDelay(DateTime.Minutes(3), function() ProduceInfantry(handofnod) end) -- DateTime.Minutes(3)
+	Trigger.AfterDelay(DateTime.Minutes(3), function() ProduceVehicle(nodairfield) end) -- DateTime.Minutes(3)
+	
+	local InitialArrivingUnits = { Actor175, Actor191, Actor192, Actor193, Actor194, Actor195, Actor196, Actor197, Actor198 }
```
`local` variables should start with a lower case letter.

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09.lua:
```diff
@@ -0,0 +1,216 @@
+		Media.PlaySpeechNotification(enemy, "Lose")
+	end)
+
+	gdiMainObjectiveBunkers = player.AddPrimaryObjective("Destroy the Nod bunkers to allow Carter's\nconvoy to pass through safely.")
```
You should use `player.AddObjective` instead (see the Lua API page). Also: Global variables should start with an upper case letter (and I would write GDI all upper case anyway).

fruestueck replied:
Can't find a AddObjective Function in the [Lua API page](https://github.com/OpenRA/OpenRA/wiki/Lua-API) but the change works.

abcdefg30 replied:
The page is for the last release, and therefore won't list new, unreleased features. (Although the top of the page shows `{DEV_VERSION}` which is indeed a bug... Fwiw, https://github.com/OpenRA/OpenRA/wiki/Lua-API-(playtest) already shows this method.)

fruestueck replied:
Should I also change the `AddSecondaryObjective` to 
`AddObjective("description", "Secondary", false)`?

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09.lua:
```diff
@@ -0,0 +1,216 @@
+end
+
+MarkCompletedObjectiveBunkers = function()
+	if not player.IsObjectiveCompleted(gdiMainObjectiveBunkers) then
```
This check is not needed. If the objective is already completed, marking it again won't make any difference. If it is failed, `MarkCompletedObjective` won't do anything either. With the check removed this entire method becomes unnecessary and can be removed (i.e. place `player.MarkCompletedObjective(gdiMainObjectiveBunkers)` directly instead of calling a function).

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09.lua:
```diff
@@ -0,0 +1,216 @@
+
+WorldLoaded = function()
+	player = Player.GetPlayer("GDI")
+	enemy = Player.GetPlayer("Nod")
```
It would be good to name those GDI and Nod respectively, but this can be left to when we port the mission to the global script.

abcdefg30 commented on mods/cnc/maps/gdi09/gdi09-AI.lua:
```diff
@@ -0,0 +1,242 @@
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+AttackPaths = { { waypoint7 }, { waypoint8 } }
```
Can we get an empty line between the end of the header and the first code line, please?

abcdefg30 commented on OpenRA.sln:
```diff
@@ -30,6 +30,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Tiberian Dawn Lua scripts",
 		mods\cnc\maps\gdi06\gdi06.lua = mods\cnc\maps\gdi06\gdi06.lua
 		mods\cnc\maps\gdi07\gdi07.lua = mods\cnc\maps\gdi07\gdi07.lua
+		mods\cnc\maps\gdi09\gdi09.lua = mods\cnc\maps\gdi09\gdi09.lua
+		mods\cnc\maps\gdi09\gdi09-AI.lua = mods\cnc\maps\gdi09\gdi09-AI.lua
```
Actually, the `-AI` script should be placed first, but this is not really an issue.

abcdefg30 reviewed (commented):
> church does not spawn a crate (should we?)

Churches leave a husk, so I don't think we should.

> A HQ (Comm center) and Nuke (Powerplant) has been added offmap

Can you change the prerequisites of `e4` and `arty` instead?

I wouldn't worry too much about the patroling and idling units too much. Your solution here looks good (although I didn't test it ingame yet). All that will change anyway once we use a proper AI in the missions.

pchote commented:
IIRC destroying a church to get the crate was a consistent feature/tactic across all the campaign missions. We should keep this, and could script it to spawn the crate next to the husk instead?

abcdefg30 commented:
We could, but don't do in any other of our TD campaign missions. I thought this was limited to RA.

fruestueck commented:
Afaik only Nod was rewarded for killing civilians (crate at church).

Sadly a fixup will take me several weeks or months as my next semster has just started. 
I would like to commit a fix (and learn), but if someone else is willing to take over that is okay for me.

abcdefg30 commented:
Ok, should be no problem. It even has the benefit that we might have the global script PR merged by then.

abcdefg30 committed:
Fix the town attackers in Infiltration not stopping

abcdefg30 committed:
Fix harvesters idling on Infiltration

abcdefg30 committed:
Fix a crash when MaxLevel of GainsExperience is zero

pchote committed:
Filter invalid actors when loading and saving games.

teinarss committed:
Add null check to EditorDefaultBrush for resources

pchote committed:
Replace actor list with count in UnitBuilderBotModule.

abcdefg30 committed:
Cache method call results in variables in Draw of LineGraphWidget

abcdefg30 committed:
Fix the XAxis of LineGraphWidget not being updated properly

abcdefg30 committed:
Fix FallsToEarth queueing an activity in its ctor

pchote committed:
Add priority levels to Exit.

pchote committed:
Add fallback exits to TS barracks.

abcdefg30 committed:
Remove $/min from the basic stats

abcdefg30 committed:
Reduce the Width of INGAME_OBSERVERSTATS_BG

abcdefg30 committed:
Reduce the width of the combat stats tab

pchote committed:
Suppress MoveIntoWorldInit for map-placed Mobile actors.

abcdefg30 committed:
Keep army and income graph disabled if they were disabled once

abcdefg30 committed:
Reduce string allocations in ObserverStatsLogic

abcdefg30 committed:
Minor style fixes

tovl committed:
pause MovePart when Mobile is paused.

abcdefg30 committed:
Remove 'yes' and 'no' in favor of 'true' and 'false'

abcdefg30 committed:
Split off a new UpdatePath for changes not going into prep-1908

abcdefg30 committed:
Add an update rule to remove 'yes' and 'no'

abcdefg30 committed:
Special case the TD spectator UI to fit the minimum width

pchote committed:
Update macOS launcher package.

This pulls in a fix for the missing libmono-native-compat.dylib

abcdefg30 committed:
Don't throw an exception when a field is missing

abcdefg30 committed:
Work around a recursive loop in TargetAndAttack

Commit:
increased ingame edge scrollspeed to 30 from 10

abc013 committed:
Use another plug location for placing plugs on a building when current plug location is already blocked

abcdefg30 committed:
Revert "Production should set SubCellInit"

This reverts commit fa1ca981ac3fb38b8ef9e85aca373964bf8331da.

abc013 committed:
Fixed zombie.shp

Added missing zombie attack frame
and other fixes.

abc013 committed:
Adjusted die sequence length of zombie

tovl committed:
Fix lastVisibleTarget not being set in FlyAttack and AttackActivity.

reaperrr committed:
Fix missing rules in prep update path

abcdefg30 committed:
Add a subCell parameter to IPositionableInfo.CanEnterCell

pchote committed:
Add UndeployOnPickup to GrantConditionOnDeploy.

pchote committed:
Allow carryall pickup orders on deployed vehicles.

abcdefg30 committed:
Let Mobile's CanEnterCell consider ToSubCell

abc013 committed:
Fix ExtractLanguageStrings utility command

abc013 committed:
Make normal difficulty on allies06b easier

abc013 committed:
Add isDead-check to the flamethrowers in allies06b

abcdefg30 committed:
Guard against overlaps on HiDPI by having a 5px border on graphs

abmyii committed:
Fix #17229: Refactor ReturnToBase.cs

reaperrr committed:
Fix zombie stand2 sequence and run tickrate

reaperrr committed:
Remove unused and buggy stand2 sequence from ants

They were unused, their settings were wrong,
and even if fixed they'd make idle ants look
glitchy, because the game switches between
stand sequences randomly.

fruestueck commented:
Comments addressed.

> I thought this was limited to RA.

Concerning the crates I think abcdefg30 is right

abmyii committed:
Prevent showing wall connections in unexplored terrain

pchote committed:
Revert "Suppress MoveIntoWorldInit for map-placed Mobile actors."

This reverts commit f0c28cc15334c7224630c375d669274f8da8fb31.

pchote committed:
Replace MoveIntoWorld with ReturnToCell/AssociateWithAirfield.

blackhand1001 committed:
Add PlaceDefenseTowardsEnemyChance trait to basebuilderbotmodule

Add PlaceDefenseTowardsEnemyChance trait to basebuilderbotmodule. This defeaults to 100 which is the current behavior. This change now allows you to set the chance that bots will place defenses evenly around the base like the AI in stock red alert and Tiberian sun did.

blackhand1001 committed:
Add UseResourceStorage trait to CashTrickler

abmyii committed:
Show dialogue only when there are unsaved changes

reaperrr committed:
Simplified initial Wanders countdown initialization

I don't see a technical reason for the old approach.

abmyii committed:
Fix #17230: Dummy audio output class

pchote committed:
Disable Carryable while submerged.

pchote committed:
Prevent movement pausing at invalid position.

pchote committed:
Don't consider unit creation as movement.

pchote committed:
Fix infantry idle animations playing immediately after creation.

reaperrr committed:
Replace AmmoPool lookup methods with properties

And gave the more suitable names while at it.
This is more in line with how we do things in
other places.

pchote committed:
Don't override spawn CenterPosition for non-aircraft reinforcements.

pchote committed:
Remove landing behaviour from force-move orders on selectable buildings.

abcdefg30 committed:
Explain that CanEnterCell ignores 'subCell' if there is a free subcell

RoosterDragon committed:
Expose a setting for Weighted A*

Replace Constants.CellCost and Constants.DiagonalCellCost with a dynamically calculated value based on the lowest cost terrain to traverse. Using a fixed value meant the pathfinder heuristics would be incorrect.

In the four default mods, the minimum cost is in fact 100, not 125. This increase would essentially allow the pathfinder to return suboptimal paths up to 25% longer in the worst case, but it would be quicker to do so.

This is exactly what Weighted A* does - overestimate the heuristic by some factor in order to speed up the search by checking fewer routes. This makes the heuristic inadmissible and it may now return suboptimal paths, but their worst case length is bounded by the weight. A weight of 125% will never produce paths more than 25% longer than the shortest, optimal, path.

We set the default weight to 25% to effectively maintain the existing, suboptimal, behaviour due to the choice of the old constant - in future it may prove a useful tuning knob for performance.

blackhand1001 committed:
Fix MCV Manager glitch when restrict building area is enabled. 

Fix MCV Manager glitch when restrict building area is enabled. It was checking if the location was close enough to the Base center instead of using the MCV Managers min and max ranges. This would cause it to often have no valid locations despite having a huge range.

blackhand1001 committed:
Simplify for loop structure now that it only has one check

Simplify for loop structure now that it only has one check

pchote committed:
Fix EjectOnDeath pilot spawning.

pchote committed:
Install x64 release into correct Program Files on Windows.

pchote committed:
Remove WorldRenderer argument from InputOverridesSelection.

pchote committed:
Force selection cursor when selection overrides input.

teinarss committed:
Fix OccupiedCells for units sharing cells

pchote committed:
Fix InputOverridesSelection only considering the closest actor.

pchote committed:
Pass contextual information to TargetOverridesSelection.

pchote committed:
Override selection if the mouse is over an already selected actor.

Commit:
Let AI ignore frozen actors and target original actors instead.

Commit:
Fix botmodules querying the Player actor before it is assigned.

pchote committed:
Introduce World and UI rendering phases.

pchote committed:
Add scissor support to IFrameBuffer.

pchote committed:
Render world via an intermediate FrameBuffer.

pchote committed:
Remove redundant zoom parameter.

Commit:
Allow GrantCondition to grant conditions permanently.

blackhand1001 committed:
Fix squadmanager adding naval units to ground attack forces.

Fix squadmanager adding naval units to ground attack forces. This was breaking the behavior of both naval and ground squads.

abcdefg30 committed:
Fix McvManagerBotModule spamming deploy orders

Removes the 'activeMCVs' list since it was not useful.
The real bugfix is not iterating over 'activeMCVs' when issueing new orders
(this was previously needed for already discovered mcvs that stopped)
but over 'newMCVs' instead.

pchote committed:
Spawn ejected pilots inside a FrameEndTask.

reaperrr committed:
Fix some header dates

Seems these were overlooked or merged after
the dates of the other files were updated.

teinarss committed:
Add IsDead check to FerryUnit OnFirstRun

reaperrr committed:
Move VocLoader to Mods.D2k

While this is strictly speaking a D2 format, any
D2 mod is pretty much guaranteed to use Mods.D2k,
so this should be a good enough place to move it to.

reaperrr committed:
Move Shp conversion/remap commands to Mods.Cnc

reaperrr committed:
Move FastByteReader from LCW to own file

And make it public.
Making it easier to move formats that depend on it
to Mods.Cnc independently later.

reaperrr committed:
Move ShpD2 and TmpRA/TD loaders to Mods.Cnc

Reducing the count of proprietary formats in Mods.Common.

Note: Moving ShpD2 to Mods.Cnc is intentional. RA/TD
use it for their original mouse shp files, and any D2(k) mod
will normally need Mods.Cnc anyway, while we can avoid
adding a Mods.D2k dependency to RA/TD this way.

reaperrr committed:
Fix FCOM still providing space while being captured

tovl committed:
Change color of selection box based on selection mode.

reaperrr committed:
Fix RA desert tree fire palette

By default WithDamageOverlay uses the actors'
palette, but RA's desert terrain uses the TD desert.pal
which isn't compatible with RA's fire animation shps.

pchote committed:
Add render_world perf sampler.

pchote committed:
Implement optimized pixel-art antialiasing mode for non-integer world zoom.

abcdefg30 committed:
Fix team chat messages not always being displayed

reaperrr committed:
Update update rule paths

reaperrr committed:
Add ProductionAirdrop.ActorType update rule

reaperrr committed:
Move ProductionAirdrop to Mods.Common

reaperrr committed:
Move WithDeliveryAnimation to Mods.Common

reaperrr committed:
Move VocLoader from D2k to Cnc

My assumption that the D2 mod would want,
let alone need Mods.D2k was wrong.
Meanwhile, it does need Mods.Cnc for at least
the Pak format as well, so this is a good enough
compromise.

pchote committed:
Add fallback exits to RA Kennel.

reaperrr committed:
Move ShpTDLoader, LZO and XORDelta formats to Mods.Cnc

They're pretty much RA/TD-specific formats.

abcdefg30 committed:
Prevent a crash in AutoTarget.HasValidTargetPriority

abcdefg30 committed:
Drop invisible invalid targets immediately when changing stance

abcdefg30 committed:
Make Parachutable public

pchote committed:
Draw voxel debug annotations using the UI renderers.

pchote committed:
Draw editor selection box using the UI renderers.

pchote committed:
Fix turret positioning in the map editor.

dragunoff committed:
Remove stray "Hotkeys" label in the Input Settings panel in TD

pchote committed:
Add zoom hotkeys.

pchote committed:
Fix hardware cursors on systems with >150% DPI scaling.

torleif committed:
fix for RA2 cloning vats

formatting

minor formatting

passing CI

closing line bracket

use production unit

opening should not be followed by a blank line

revert tab change

pchote committed:
Overhaul cursor double setting.

pchote committed:
Add VSync setting.

pchote committed:
Account for game ticks when applying the frame limiter.

pchote committed:
Disable frame limiter by default

VSync does a more reliable job of this.

pchote committed:
Replace frame limit text field with a slider.

reaperrr committed:
Remove duplication between IMA and AUD readers

These parts were identical duplicates, so I removed them
and made AudReader depend on ImaAdpcmReader instead.

Applied some style fixes while I was at it.

reaperrr committed:
Remove VqaReader dependency on AudReader

Use ImaAdpcmReader directly, since VQAs only contain
header-less raw IMA  ADPCM audio.

reaperrr committed:
Move AUD support to Mods.Cnc

pchote committed:
Draw GPS dots using the UI renderers.

pchote committed:
Fix ATEK vision ranges.

pchote committed:
Add MinRange support to AffectsShroud.

pchote committed:
Remove overlapping vision ranges from RA actors.

This brings a significant perf saving by reducing
the number of evaluated tiles.

dragunoff committed:
Remove "Surrender" button for single player games

pchote committed:
Implement new viewport size/zoom UI.

abcdefg30 committed:
Move unit order resolving into a helper function

abcdefg30 committed:
Remove a nowadays unnecessary loop safeguard

abcdefg30 committed:
Fix a potential crash in PlayVoiceForOrders

abcdefg30 committed:
Move voice playing for orders into a helper function

pchote committed:
Restore NukePower support for proxy actors.

abmyii committed:
Fix for mod credits overriding engine credits

pchote committed:
Fix WithTextDecoration rendering.

pchote committed:
Fix duplicated/distorted SequencePlaceBuildingPreview annotations.

pchote committed:
Fix NV shader compile errors in combined.frag.

pchote committed:
Fix ModContent cursor.

abcdefg30 committed:
Remove ISync from RallyPoint

pchote committed:
Fix MiniYaml parsing of empty comments

abcdefg30 committed:
Fix an indentation style issue in TransformsIntoAircraft

pchote committed:
Rework chrome.yaml format and panel rendering.

pchote committed:
Manual chrome.yaml cleanups.

pchote committed:
Fix incorrect region definitions.

abcdefg30 committed:
Pass DamageTypes in Explodes

pchote committed:
Filter invalid actor IDs when restoring selection save data.

Commit:
Adaption to global campaign script
Fixed changed rules.yaml Prerequisities for Units (ARTY, LTNK, E4)

fruestueck commented:
unintedned rebase of 400 bleed commits.
